cmd/xl-v1-multipart.go

```diff
@@ -577,32 +587,34 @@ func (xl xlObjects) CompleteMultipartUpload(ctx context.Context, bucket string,
 	}
 	defer uploadIDLock.Unlock()
 
-	if !xl.isUploadIDExists(bucket, object, uploadID) {
-		return oi, errors.Trace(InvalidUploadID{UploadID: uploadID})
+	if !xl.isUploadIDExists(ctx, bucket, object, uploadID) {
+		logger.LogIf(ctx, InvalidUploadID{UploadID: uploadID})
+		return oi, InvalidUploadID{UploadID: uploadID}
 	}
 
 	// Check if an object is present as one of the parent dir.
 	// -- FIXME. (needs a new kind of lock).
-	if xl.parentDirIsObject(bucket, path.Dir(object)) {
-		return oi, toObjectErr(errors.Trace(errFileAccessDenied), bucket, object)
+	if xl.parentDirIsObject(ctx, bucket, path.Dir(object)) {
+		logger.LogIf(ctx, errFileAccessDenied)
```

Comment:
No need to log here since `parentDirIsObject` already takes care or is supposed to take care of the `FileAccessDenied` error condition.

Reply:
done